test/common_nn.py

```diff
@@ -854,6 +856,22 @@ def test_cuda(self, test_case):
                 for cpu_d_p, gpu_d_p in zip(cpu_param[1], gpu_param[1]):
                     test_case.assertEqual(cpu_d_p, gpu_d_p, 2e-4)
 
+            # Run double-backwards on CPU and GPU and compare results
+            # TODO: BatchNorm double-backwards is currently broken, see
+            # #4422; remove this when it is fixed.
+            if self.check_gradgrad and not self.get_name().startswith('test_BatchNorm'):
+                for i in range(5):
+                    cpu_output_t = cpu_output.data if isinstance(cpu_output, Variable) else cpu_output
+                    cpu_gradOutput = Variable(cpu_output_t.clone().bernoulli_(), requires_grad=True)
+                    gpu_gradOutput = Variable(cpu_gradOutput.type('torch.cuda.FloatTensor').data, requires_grad=True)
+                    cpu_gradInput = test_case._backward(cpu_module, cpu_input, cpu_output, cpu_gradOutput, create_graph=True)
+                    gpu_gradInput = test_case._backward(gpu_module, gpu_input, gpu_output, gpu_gradOutput, create_graph=True)
+                    cpu_input.grad.sum().backward(retain_graph=True)
```

Comment:
I think this should use `torch.autograd.grad` and not accumulate back into the parameter gradients.

Reply:
For the record, making this change helped uncover some other bugs with double backwards (#4500). So, :+1: